packages/sources/coinmetrics/src/endpoint/burned.ts

```diff
@@ -34,30 +60,48 @@ export const execute: ExecuteWithConfig<Config> = async (request, _, config) =>
 
   const jobRunID = validator.validated.id
   const asset = validator.overrideSymbol(AdapterName, validator.validated.data.asset)
-  const frequency = validator.validated.data.frequency || DEFAULT_FREQUENCY
-  const pageSize = validator.validated.data.pageSize || DEFAULT_PAGE_SIZE
-  const url = 'timeseries/asset-metrics'
-  const metrics = 'FeeTotNtv,RevNtv,IssTotNtv'
+  const startDate = validator.validated.data.startDate
+  const endDate = validator.validated.data.endDate
 
   const params = {
     assets: (asset as string).toLowerCase(),
-    metrics,
-    frequency,
+    metrics: METRICS,
+    frequency: FREQUENCY,
+    page_size: config.pageSize,
+    start_time: startDate,
+    end_time: endDate,
     api_key: config.apiKey,
-    page_size: pageSize,
   }
 
-  const options = { ...config.api, params, url }
+  const options = { ...config.api, params, url: URL }
+
+  let totalBurnedTKN = BigNumber.from('0')
+  let response
+  /*eslint no-constant-condition: ["error", { "checkLoops": false }]*/
+  while (true) {
+    response = await Requester.request<ResponseSchema>(options)
```

Comment:
I don't think we should be doing this loop here, it's better fit for the total burned endpoint.